tests/test_maliar.py

```diff
@@ -708,3 +761,89 @@ def test_block_value_net():
 
     # Should give same results
     assert torch.allclose(values, values2)
+
+
+class TestEulerResidualsBenchmarks(unittest.TestCase):
```

Comment:
"TestEulerResidualBenchmarks" makes it sound like this will be using one of the benchmark models and using it to test the functions.

Both the two currently included test basically just check to make sure the code doesn't crash, and also that it doesn't return zero residuals for a wrong decision rule.

But what could test that the residual function is actually computing the residual accurately?

What about a test that uses the benchmarks, and computes the Euler residuals for the optimal analytically known solution?